gluten-core/src/main/scala/io/glutenproject/extension/RewriteSparkPlanRulesManager.scala

```diff
@@ -58,13 +58,19 @@ class RewriteSparkPlanRulesManager(rewriteRules: Seq[Rule[SparkPlan]]) extends R
     }
   }
 
-  private def getTransformHintBack(origin: SparkPlan, rewrittenPlan: SparkPlan): TransformHint = {
+  private def getTransformHintBack(
+      origin: SparkPlan,
+      rewrittenPlan: SparkPlan): Option[TransformHint] = {
     // The rewritten plan may contain more nodes than origin, here use the node name to get it back
     val target = rewrittenPlan.collect {
       case p if p.nodeName == origin.nodeName => p
     }
     assert(target.size == 1)
-    TransformHints.getHint(target.head)
+    if (TransformHints.isTransformable(target.head)) {
+      None
+    } else {
+      Some(TransformHints.getHint(target.head))
+    }
```

Comment:
do we need this change ? it seems verbose than before

Reply:
The code looks simpler before, but implicitly hid the fact that the rule's logic depends on `TRANSFORM_SUPPORTED` which is considered unreliable after this patch.

The if/else just pulled the unreliable logic closer to API calls, say, to call `isTransformable` which should be finally deprecated and removed. So it becomes easier for further code cleanups because of this change.

Reply:
Can refer to the code comment https://github.com/oap-project/gluten/pull/4731/files/3f03b483e23d8b2f64a924ab473b1630b95d4379#diff-af786f23393cb8ac3fb3018ea6062722ce6d7388b1a69d3cd512ac7f1ab69435R71-R77

Reply:
thank you for the explanation